portfolio/src/main/java/com/google/sps/servlets/Comment.java

```diff
@@ -109,33 +118,45 @@ public Entity createEntity() {
     commentEntity.setProperty("lastName", this.lastName);
     commentEntity.setProperty("email", this.email);
     commentEntity.setProperty("comment", this.comment);
+    commentEntity.setProperty("timestamp", this.timestamp);
     commentEntity.setProperty("visitReason", this.visitReason.name()); // set to enum property name
 
+    // Return the created entity
     return commentEntity;
   }
 
   /**
    * Static method to convert a datastore to an ArrayList of Comment objects.
    * @param datastore instance of DatastoreService that contains "Comment" entities
+   * @param max maximum number of comments to return. Set to 0 for no maximum
    * @return an ArrayList of comment objects with data from each entity in datastore.
    */
-  public static ArrayList<Comment> datastoreToArrayList(DatastoreService datastore) {
+  public static ArrayList<Comment> datastoreToArrayList(DatastoreService datastore, int max) {
     // Create a query
-    Query query = new Query("Comment");
+    Query query = new Query("Comment")
+        .addSort("timestamp", Query.SortDirection.DESCENDING);
 
     // Get the results from the query
-    PreparedQuery results = datastore.prepare(query);
+    PreparedQuery resultsQuery = datastore.prepare(query);
```

Comment:
Maybe use [FetchOptions](https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/FetchOptions) to avoid the need of that early stop at lines 152-153

Reply:
Clever - this is really helpful.